### DISCUSSION TITLE
Changing csv format

### DISCUSSION THREAD

**Discussion by LeratLy** [immersivecognition/unity-experiment-framework#99] [Q&A] [answered]
Hello! I am currently collecting data with this great framework but unfortunately it saves it seperated with ';' and floats with ',' in a csv. I would prefer a data collection with ',' instead of ',' and '.' instead of ',' because I want to analyse it computationally. I read that this is becaue I use it from a non english location but where can I change this behavior in the code? Couldn't finde an answer yet. Greetings

**Comment by jackbrookes** [comment, marked as answer]
Hi! I think the most straightforward way is to change the language / localisation of the PC you are running Unity on to English. 

Secondly I imagine most CSV reading software (Python, R, MATLAB) supports using delimiters of `;`. Commas are not necessary.

Another way (untested) is to change the locale of your program using code. For example, you can write a simple script that forces to English locale on Awake() using something this, and attach it to an object in your scene:

```cs
using UnityEngine;
using System.Globalization;
using System.Threading;

public class CultureSetter : MonoBehaviour
{
    private void Awake()
    {
        Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US");
    }
}
```

**Reply by LeratLy** [reply]
Thanks! I tried the last approach and it worked out fine.

**Reply by eioe** [reply]
FYI, we have also used something similar to the latter approach in a couple of projects and it has never produced problems this far (OS: German Win10). 

@jackbrookes Idk, how common this problem it is and how specific to UXF usage. But maybe, some suffering could be avoided (talking about my former self) by including a warning regarding this somewhere in the wiki or documentation, saying something like: "Watch out if you're using UXF on a machine that runs a non-english OS: your data might get saved using `,` as decimal separator." 
When running into this problem for the first time, we had developped and piloted on an English-OS machine (everything fine) but when we ran it on the PC in the lab  (German-OS) for actual data acquisition, we collected a bunch of data where decimal separators were undistinguishable from (column) separators (both `,`) before we realized.
I'm not suggesting to include this in the wiki/docu because I think this is a flaw of UXF, rather because I see its sites becoming a well-frequented hub of knowledge exchange where such a warning could efficiently reach many people who might run into the same issue.

**Reply by jackbrookes** [reply]
thanks! I will take note.

**Reply by jackbrookes** [reply]
In the latest UXF update I made an option to force en-US locale. This will be enabled by default.
